core/src/main/java/org/elasticsearch/env/Environment.java

```diff
@@ -138,9 +145,32 @@ public Environment(Settings settings) {
                 dataFiles[i] = PathUtils.get(dataPaths.get(i));
                 dataWithClusterFiles[i] = dataFiles[i].resolve(clusterName.value());
             }
+            if (DEFAULT_PATH_DATA_SETTING.exists(settings)) {
+                final String defaultPathDataValue = DEFAULT_PATH_DATA_SETTING.get(settings);
+                final Set<Path> dataFilesSet = Arrays.stream(dataFiles).collect(Collectors.toSet());
+                final Path defaultPathData = PathUtils.get(defaultPathDataValue);
+                if (dataFilesSet.size() == 1 && dataFilesSet.contains(defaultPathData)) {
+                    // default path data was used to set path data
+                    this.defaultPathData = null;
+                } else if (dataFilesSet.contains(defaultPathData)) {
+                    final String message = String.format(
+                            Locale.ROOT,
+                            "do not include default.path.data [%s] in path.data %s",
```

Comment:
Yes, I understand, what I meant is that it's entirely okay for `default.path.data=/var/lib/elasticsearch` and `path.data=/var/lib/elasticsearch`, it's *not* okay for `default.path.data=/var/lib/elasticsearch` and `path.data=/var/lib/elasticsearch,/mnt/data2` (where path.data is an array), I didn't want the message to imply that `path.data` can *never* include `default.path.data`